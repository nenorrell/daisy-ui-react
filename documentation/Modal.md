# Modal
A simple modal component that can be used for displaying data, alerts, etc.

For Modal styling specifics, please see: https://daisyui.com/components/modal/

## Props

| Name | Type | Required | Default | Description |
|------|------|----------|---------|-------------|
| `id` | string | optional | - | The id of the modal |
| `isOpen` | boolean | required | - | Whether the modal is open or not |
| `closeHandler` | function | required | - | Function to be called when the modal is closed |
| `variant` | string | optional | `"neutral"` | The color variant of the modal |
| `position` | string | optional | `"middle"` | The position of the modal |
| `className` | string | optional | - | Additional className to be added to the modal |
| `cornerClose` | boolean | optional | `false` | Whether the modal should have a close button in the corner or not |
| `overlayClose` | boolean | optional | `true` | Whether the modal should be closed when clicking on the overlay or not |
| `noActionButton` | boolean | optional | `false` | Whether the modal should have a close button in the action section or not |
| `closeButtonVariant` | string | optional | - | The color variant of the close button |
| `closeButtonText` | string | optional | "Close" | The text of the close button |
| `noActionButton` | boolean | optional | - | Whether the modal should not have an action section |

## Example

```javascript
import { Modal } from 'daisy-ui-react-components';

function App() {
  const [showModal, setShowModal] = useState(false);
  const handleConfirm = () => {
    // handle confirmation logic
  };

  return (
    <div>
      <button onClick={() => setShowModal(true)}>Show Confirm Modal</button>
      <Modal 
        isOpen={showModal} 
        closeHandler={() => setShowModal(false)} 
      >
        Hello!
      </Modal>
    </div>
  );
}
```
