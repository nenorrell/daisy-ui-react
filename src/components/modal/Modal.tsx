import { forwardRef, PropsWithChildren } from "react";
import { BaselessVariant } from "../../@types/Colors";
import { Button } from "../buttons/Button";
import { IModalFrame, ModalFrame } from "./ModalFrame";

interface IModal extends IModalFrame{
    closeButtonVariant ?:BaselessVariant
    closeButtonText ?:string
    noActionButton ?:boolean
}
export const Modal = forwardRef<HTMLDivElement, PropsWithChildren<IModal>>((
    props,
    ref
)=>{
    return (
        <ModalFrame
            ref={ref}
            id={props.id}
            variant={props.variant}
            isOpen={props.isOpen}
            position={props.position}
            className={props.className}
            closeHandler={props.closeHandler}
            cornerClose={props.cornerClose}
            overlayClose={props.overlayClose}
        >
            {props.children}

            { !props.noActionButton &&
                <div className="modal-action">
                    <Button variant={props.closeButtonVariant} onClick={props.closeHandler}>
                        {props.closeButtonText || "Close"}
                    </Button>
                </div>
            }
        </ModalFrame>
    );
});
