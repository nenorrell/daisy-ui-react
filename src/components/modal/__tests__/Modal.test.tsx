import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Modal } from "../Modal";

describe("Modal", () => {
    it("should render with default values", () => {
        const { container } = render(<Modal isOpen={true} />);
        expect(container).toMatchSnapshot();
    });

    it("should call closeHandler when close button is clicked", () => {
        const closeHandler = jest.fn();
        const { container } = render(<Modal isOpen={true} closeHandler={closeHandler} />);
        fireEvent.click((container as any).querySelector(".modal-action button"));
        expect(closeHandler).toHaveBeenCalled();
    });

    it("should not render with confirm button if noActionButton is true", () => {
        const { container } = render(<Modal isOpen={true} noActionButton={true} />);
        expect(container.querySelector(".modal-action")).toBeFalsy();
    });
});
