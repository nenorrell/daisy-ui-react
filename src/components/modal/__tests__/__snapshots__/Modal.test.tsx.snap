// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render with default values 1`] = `
<div>
  <div
    class="modal modal-open"
  >
    <div
      class="modal-box bg-neutral text-neutral-content"
    >
      <div
        class="modal-action"
      >
        <button
          class="btn btn-neutral btn-md"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
